ddtrace/contrib/dbapi/__init__.py

```diff
@@ -69,10 +69,13 @@ def __enter__(self):
 class TracedConnection(wrapt.ObjectProxy):
     """ TracedConnection wraps a Connection with tracing code. """
 
-    def __init__(self, conn):
+    def __init__(self, conn, pin=None):
         super(TracedConnection, self).__init__(conn)
         name = _get_vendor(conn)
-        Pin(service=name, app=name).onto(self)
+        if pin and pin.enabled():
```

Comment:
if we pass a disabled `pin`, we'll create a new one that is enabled. Not sure it's the right thing to do. There is an edge case that you're trying to address?

Also what if we write this code as (in case `enabled()` is not required):
```python
name = _get_vendor(conn)
db_pin = pin or Pin(service=name, app=name, app_type="db")
db_pin.onto(self)
```

Reply:
No particular reason. I'll use that.